pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -37,16 +37,33 @@ class ModificationsExportDialog extends React.Component {
         super(props);
 
         this.state = {
-            active_tab: "shell"
+            active_tab: "shell",
+            copied: false
         };
 
         this.handleSelect = this.handleSelect.bind(this);
+        this.getScript = this.getScript.bind(this);
     }
 
     handleSelect(active_tab) {
         this.setState({ active_tab });
     }
 
+    getScript() {
+        try {
+            navigator.clipboard.writeText(this.props[this.state.active_tab])
+                    .then(() => {
+                        this.setState({ copied: true });
+                        setTimeout(() => {
+                            this.setState({ copied: false });
+                        }, 3000);
+                    })
+                    .catch(e => console.error('Text could not be copied: ', e ? e.toString() : ""));
+        } catch (error) {
+            console.error('Text could not be copied: ', error.toString());
```

Comment:
Why does this need to be handled twice?

Do we expect this to fail in some scenarios, like particular browsers? If so, do we need to change the button then (perhaps with a red X or so) to indicate the failure? Or is that a Should Not Happen™ case?

Reply:
One catches when this API is not supported by browser and second one if there are no permissions.
I think this is rather edge case (copying to clipboard works, reading clipboard is usually the problematic one) but surely it could happen. Should Not Happen™ sounds about right, but I can handle it in the UI. @andreasn what do you think and if you think this should be visually shown, what would be the right UI?

Reply:
Thanks. Indeed if writing clipboard doesn't ask, then it's fine IMHO.

Reply:
I think it's fine to only show it in the console if it's so rare.